### PR TITLE
Sync versions.props in project folders

### PR DESCRIPTION
**Pull request by korser1**
Proposal, for review only

https://github.com/SteeltoeOSS/Samples/issues/136

### CODE REVIEW

**Comment by TimHess** [comment]
Thanks for getting this work started @korser1. 

Rather than a collection of `versions.props` files that need to be imported into every single project file, I think we should eliminate the need for project file updates by using a convention that is recognized by MSBuild - [Directory.Build.props](https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build) should work now, or else [Directory.Packages.props](https://github.com/NuGet/Home/wiki/Centrally-managing-NuGet-package-versions) _may_ work now but could be an even better solution at some point.

The other portion of the issue, which you have taken a stab at is updating of the `.props` copies. I suppose a brute force process/script that duplicates a root properties file into any folder that contains a .sln should work, but it would be more elegant if we selectively updated versions and had targeted version properties so the files themselves could be smaller... but that could also be done in the future if the brute force process causes issues so I think I'm ok with the current approach

**Comment by korser1** [comment]
> Thanks for getting this work started @korser1.
> 
> Rather than a collection of `versions.props` files that need to be imported into every single project file, I think we should eliminate the need for project file updates by using a convention that is recognized by MSBuild - [Directory.Build.props](https://docs.microsoft.com/en-us/visualstudio/msbuild/customize-your-build) should work now, or else [Directory.Packages.props](https://github.com/NuGet/Home/wiki/Centrally-managing-NuGet-package-versions) _may_ work now but could be an even better solution at some point.
> 
> The other portion of the issue, which you have taken a stab at is updating of the `.props` copies. I suppose a brute force process/script that duplicates a root properties file into any folder that contains a .sln should work, but it would be more elegant if we selectively updated versions and had targeted version properties so the files themselves could be smaller... but that could also be done in the future if the brute force process causes issues so I think I'm ok with the current approach

@Tim, thank you for your reply.
I looked into Directory.Build.props, but decided no to touch them, as this file could be introduced inside the specific project and sync script will override it. Ok, will try to replace versions.props

Regarding the packages props, I agree with you, but the feature is in the review, And should see how it will work with Dockerfile with accessing, for example.

File: copy_props.ps1 [hunk 1]
```diff
@@ -0,0 +1,25 @@
+###
```
Comment (ccheetham) [inline copy_props.ps1:1]: See https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_comment_based_help?view=powershell-7.2 for adding help to powershell scripts.

You can rewrite this comment/help section like:
```
<#
.Synopsis
Configure Sample projects build properties (Directory.Build.props).
#>
```

And then from a powershell cli run:
```
PS > get-help ./copy_props.ps1

NAME
    /Users/ccheetham/src/github.com/SteeltoeOSS/Samples/copy_props.ps1

SYNOPSIS
    Configure Sample projects build properties (Directory.Build.props).
...
```
```

File: copy_props.ps1 [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+### Sync Directory.Build.props in project folder with "main" config/Directory.Build.props file
+### Project should reference to any Steeltoe package with Version=$(SteeltoeVersion).
+### 
+### TODO: add shell script for Linux
```
Comment (ccheetham) [inline copy_props.ps1:5]: Powershell is available for Linux so this TODO is effectively completed.

You can make it an executable script on Linux by:
* converting file line endings from DOS to UNIX
* set the file's execute bit (may not be able to do this from Windows?)
* making the first line of the script:
```
#/usr/bin/env pwsh
```

If you don't have access to a Linux, or are uncomfortable making these mods, I can make them after the PR is accepted.

**Review by ccheetham — CHANGES_REQUESTED** [review]
Looks good.  I've requested some mods to the copy_props.ps1 script.  Specifics in it's feedback.

* rename to configure.ps
* convert comments to help comments
* (optional) make script executable on Linux

**Comment by korser1** [comment]
> Looks good. I've requested some mods to the copy_props.ps1 script. Specifics in it's feedback.
> 
> * rename to configure.ps
> * convert comments to help comments
> * (optional) make script executable on Linux

Done

**Review by ccheetham — APPROVED** [review]

**Pull request merged**